docs/example_nodejs.md

```diff
@@ -1,34 +1,314 @@
+<figure>
+  <a class="is-image" title="Result of clicking the button  on the “deploy” resource" href="https://cloud.tilt.dev/snapshot/todo">
+    <img src="assets/docimg/example-nodejs-naive-deploy.png">
+  </a>
+  <figcaption>Clicking the button triggers the "deploy" local_resource, which in turn kicks off an update to the server. (Click the screenshot to see an interactive view.)</figcaption>
+</figure>
 
-For now, check out:
+| Approach | Deploy Time (after initial)
+|---|---|
+| Naive | 11.31-14.21s |
+{:.benchmark-report}
 
-- [abc123](https://github.com/windmilleng/abc123) a mini microservice app with a simple HTTP NodeJS server called `letters`.
-- The [NodeJS microservice with Hot Reloading](nodejs_microservice_hotreloading.html) tutorial for using `live_update`.
-- Servantes (below), our microservice app in multiple languages.
+If you look closely, the elapsed time displayed in the Tilt sidebar is different
+than the benchmark our app logged. That's OK! In multi-service development,
+there are many benchmarks we care about---the time to build the image, the time
+to schedule the process, and the time until the server is ready to serve
+traffic.
 
+The Tilt sidebar gives you some default benchmarks, _and_ the tools to capture
+your own benchmarks.
 
-## Servantes
+Our benchmarks show this is slow. Can we do better?
 
-[Servantes](https://github.com/windmilleng/servantes) is a personalized homepage
-app implemented with many microservices in multiple languages. You can run it
-with the commands
+## Step 2: Why Is the Docker Build So Slow?
 
+The first thing I notice when I click "deploy" is a bunch of logs from `yarn install`; and not just once, but _every dang time_. This is a hint that we can optimize our Dockerfile to be smarter about caching. With a little rearranging, our [new Dockerfile]((https://github.com/windmilleng/tilt-example-nodejs/blob/master/2-optimized-dockerfile/Dockerfile)) looks like this:
 ```
-git clone https://github.com/windmilleng/servantes
-cd servantes; tilt up
+FROM node:10
+
+WORKDIR /app
+
+ADD package.json .
+ADD yarn.lock .
+RUN yarn install
+
+ADD . .
+
+ENTRYPOINT [ "node", "/app/index.js" ]
 ```
 
-Each widget is implemented by a different microservice backend.
+Here's what it looks like when we build with our new Dockerfile:
+
+<figure>
+  <a class="is-image" title="Result of clicking the button  on the “deploy” resource" href="https://cloud.tilt.dev/snapshot/todo">
+    <img src="assets/docimg/example-nodejs-optimized-dockerfile.png">
+  </a>
+  <figcaption>"RUN pip install..." now uses the cache instead of actually running a long, slow command. (Click the screenshot to see an interactive view.)</figcaption>
+</figure>
+
+Hooray, we're now using the cache instead of running `yarn install` for every single build. (For more on the principles at work here, [check out this guide](http://bitjudo.com/blog/2014/03/13/building-efficient-dockerfiles-node-dot-js/).)
```

Comment:
I would probably point to something an actively-maintained guide like https://nodejs.org/en/docs/guides/nodejs-docker-webapp/, rather than a blog post. The blog post will go out of date. We don't expect the author to go back and fix it. Whereas a widely-used guide gets more traffic and fixes.

Reply:
that's a fair point but the blog post is really precisely about the concept I'm alluding to here, whereas you need to dig for it in the nodejs guide. (also the nodejs guide links to the same blogpost...)